### PR TITLE
Adding dev artifact

### PR DESCRIPTION
**Pull request by JenMadiedo**
Issue Number: 425

## Summary
Removing manual api.yaml and adding CICD steps to generate dev artifact

## Changes

- List of comprehensive changes

## Checklist

- [ ] Local Tests Passing?
- [ ] CICD and Pipeline Tests Passing?
- [ ] Added any new Tests?
- [ ] Documentation Updates Made?
- [ ] Are there any API Changes? If yes, please describe below.
- [ ] This is not a breaking change. If it is, please describe it below.

## Are there API Changes?
If yes, what are the expected API Changes? Please link to an API-Comparison workflow with the API Diff.

## Is this a breaking change?
If yes, what workflow does this break? 

## Anything else?
Other comments, collaborators, etc.
This PR Closes Issue #

### CODE REVIEW

File: .github/workflows/build-webapi.yaml [hunk 1]
```diff
@@ -55,15 +55,46 @@ jobs:
...
       
-      - name: Upload artifact
+      - name: Upload branch artifact
         uses: actions/upload-artifact@v1
         with:
           name: pr-swagger.yaml
-          path: src/CarbonAware.WebApi/src/api/v1/swagger.yaml
+          path: src/CarbonAware.WebApi/src/api/v1/pr-swagger.yaml
+
+      - name: Checkout Dev Branch
```
Comment (juzuluag) [inline .github/workflows/build-webapi.yaml]: Shouldn't be these steps a different job?

Comment (dnastrain) [reply .github/workflows/build-webapi.yaml]: I agree with @juzuluag -- the better place to do this, is in the "api-comparison" job, which already checks-out the dev branch, and already builds a "dev-branch-based-swagger" file from it -- and then downloads the "PR-branch-based-swagger" file from the prior job -- I would recommend uploading the additional `dev-swagger.yaml` build-artifact after the "Generate Open API" step in the "api-comparison" job

**Review by dnastrain — APPROVED** [review]
LGTM , :shipit:

**Review by juzuluag — APPROVED** [review]
👍

**Pull request merged**